lib/fluent/plugin/in_tail.rb

```diff
@@ -440,6 +477,9 @@ def stop_watchers(targets_info, immediate: false, unwatched: false, remove_watch
           end
         end
       }
+      if @thread_pool
+        @thread_pool.stop
```

Comment:
Please consider adding `@thread_pool = nil` to clarify that `@thread_pool`never reused.

Reply:
Fixed in https://github.com/fluent/fluentd/pull/3185/commits/d15eae9304651c4f888355496ca08b0c9db24f60.